assets/js/common/PatchList/PatchList.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import {
+  EOS_SHIELD_OUTLINED,
+  EOS_CRITICAL_BUG_OUTLINED,
+  EOS_ADD_BOX_OUTLINED,
+} from 'eos-icons-react';
+import { format as formatDate } from 'date-fns';
+import classNames from 'classnames';
+
+import Table from '@common/Table';
+import { computedIconCssClass } from '@lib/icon';
+
+const iconFromAdvisoryType = (
+  advisoryType,
+  centered = false,
+  hoverOpacity = true,
+  size = 'l'
+) => {
+  const hoverOpacityClassName = {
+    'hover:opacity-75': hoverOpacity,
+    'hover:opacity-100': !hoverOpacity,
+  };
+
+  switch (advisoryType) {
+    case 'security_advisory':
+      return (
+        <EOS_SHIELD_OUTLINED
+          className={classNames(
+            hoverOpacityClassName,
+            computedIconCssClass('fill-red-500', centered)
+          )}
+          size={size}
+        />
+      );
+    case 'bugfix':
+      return (
+        <EOS_CRITICAL_BUG_OUTLINED
+          className={classNames(
+            hoverOpacityClassName,
+            computedIconCssClass('fill-yellow-500', centered)
+          )}
+          size={size}
+        />
+      );
+    case 'enhancement':
+      return (
+        <EOS_ADD_BOX_OUTLINED
+          className={classNames(
+            hoverOpacityClassName,
+            computedIconCssClass('fill-yellow-500', centered)
+          )}
+          size={size}
+        />
+      );
+    default:
+      return null;
+  }
+};
+
+const patchListConfig = {
+  usePadding: false,
+  pagination: true,
+  columns: [
+    {
+      title: 'Type',
+      key: 'advisory_type',
+      render: (content, _) => iconFromAdvisoryType(content),
+    },
+    {
+      title: 'Advisory',
+      key: 'advisory_name',
+      render: (content, item) => (
+        <Link className="text-jungle-green-500 hover:opacity-75" to={item.url}>
+          {content}
+        </Link>
+      ),
```

Comment:
Move this to `onNavigate` like other components do. This way we don't depend directly from `react-router-dom` and we can test the component without wrapping it in a `MemoryRouter` :+1:

Reply:
@dottorblaster I changed the patches data, so it can hold an onNavigate function.

```jsx
patches = [
  {
    ....
    onNavigate = () => navigate(`/patch/${id}`)
  }
]

<PatchList patches={patches} />
```

The other alternative would have been to do something like this:

```jsx
<PatchList patches={[{...}]} onNavigage = {(patch) => navigate(`/patch/${patch.id}`)} />
```

I think I prefer the first solution, because it leaves more flexibility, but the latter API is cleaner. What do you think?

Reply:
@janvhs usually we pass an onNavigate function to the whole component and then encapsulate the logic inside, since it's really something tied to the rendering layer. This way we can compose the data we have from e.g. the Redux store or whatever external source with the actual implementation detail of the rest of the frontend.

The on-click navigation is really something that only matters for the rendering.